src/tests/functional/frame_tests.ts

```diff
@@ -15,6 +15,18 @@ export class FrameTests extends FunctionalTestCase {
     this.assert.equal(await frame.getAttribute("data-loaded-from"), currentPath)
   }
 
+  async "test a frame whose src references itself does not infinitely loop"() {
+    await this.clickSelector("#frame-self")
+
+    await this.nextEventNamed("turbo:before-fetch-response")
+
+    this.assert.equal(await this.getVisibleText("h2"), "Frames: Self")
```

Comment:
Not sure I agree with this assertion… I think maybe we ought to avoid loading frames when we know they reference themselves, and log a specific error message explaining the problem.